File: Models/Transaction.cs
using System.Text.Json.Serialization;

namespace ArcadianTestTransactionApp.Models
{
    public class Transaction
    {
        [JsonPropertyName("transactionId")]
        public int TransactionId { get; set; }

        [JsonPropertyName("transactionName")]
        public string TransactionName { get; set; }

        [JsonPropertyName("transactionDate")]
        public DateTime TransactionDate { get; set; }

        [JsonPropertyName("transactionCost")]
        public double TransactionCost { get; set; }
    }
}

File: ClientApp/src/app/models/transaction.ts
export class Transaction {
  transactionId: number;
  transactionName: string;
  transactionDate: Date;
  transactionCost: number;

  constructor() {
    this.transactionId = 0;
    this.transactionName = '';
    this.transactionDate = new Date();
    this.transactionCost = 0;
  }
}
